src/lib/components/PrismicImage.svelte

```diff
@@ -1,34 +1,63 @@
 <script lang="ts">
-	import * as prismicT from "@prismicio/types";
+	import * as prismicH from "@prismicio/helpers"
+	import type * as prismicT from "@prismicio/types";
+	
+	/**
+	 * A Prismic Image field.
+	 */
 	export let field: prismicT.ImageField;
-	let className: string;
 
-	// creates a `class` property, even
-	// though it is a reserved word
-	export { className as class };
+	/**
+	 * An object of Imgix URL API parameters to transform the image.
+	 *
+	 * See: https://docs.imgix.com/apis/rendering
+	 */
+	export let imgixParams: Parameters<typeof prismicH.asImageSrc>[1] | null | undefined = null;
 
-	const usePrismicImage = (
-		field: prismicT.ImageField,
-	): {
-		url: string;
-		width?: number;
-		height?: number;
-		alt?: string;
-		copyright: string | undefined;
-	} => {
-		field.url;
-		return {
-			url: field.url || "",
-			alt: field.alt || "",
-			width: field.dimensions?.width || undefined,
-			height: field.dimensions?.height || undefined,
-			copyright: field.copyright || undefined,
-		};
-	};
+	/**
+	 * Widths used to build a `srcset` value for the Image field.
+	 *
+	 * If a `widths` prop is not given or `"defaults"` is passed, the
+	 * following widths will be used: 640, 750, 828, 1080, 1200, 1920, 2048,
+	 * 3840.
+	 *
+	 * If the Image field contains responsive views, each responsive view
+	 * can be used as a width in the resulting `srcset` by passing
+	 * `"thumbnails"` as the `widths` prop.
+	 */
+	export let widths: NonNullable<Parameters<typeof prismicH.asImageWidthSrcSet>[1]>["widths"] | null | undefined = null;
```

Comment:
`widths="defaults"` should be a valid value representing the default widths (`640, 750, 828, 1080, 1200, 1920, 2048, 3840`).

```ts
export let widths: NonNullable<Parameters<typeof prismicH.asImageWidthSrcSet>[1]>["widths"] | "defaults" | undefined = null;
```

`"defaults"` is necessary to support both `widths` and `pixelDensities` (see the next comment re: `pixelDensities`).

Reply:
✅

Reply:
✅